test/e2e/xtx/manager.go

```diff
@@ -0,0 +1,99 @@
+package xtx
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"math/big"
+	"strings"
+
+	"github.com/omni-network/omni/contracts/bindings"
+	"github.com/omni-network/omni/lib/errors"
+	"github.com/omni-network/omni/test/e2e/netman"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+)
+
+type TxSenderManager struct {
+	txSender map[uint64]Sender
+	abi      *abi.ABI
+}
+
+type XCallArgs struct {
+	DestChainID uint64
+	Address     common.Address
+	Data        []byte
+	GasLimit    uint64
+}
+
+func New(ctx context.Context, portals map[uint64]netman.Portal, privateKey *ecdsa.PrivateKey) (TxSenderManager, error) {
+	// create ABI interface
+	parsedAbi, err := abi.JSON(strings.NewReader(bindings.OmniPortalMetaData.ABI))
+	if err != nil {
+		return TxSenderManager{}, errors.Wrap(err, "parse abi error")
+	}
+
+	// create tx sender manager
+	manager := TxSenderManager{
+		txSender: make(map[uint64]Sender),
+		abi:      &parsedAbi,
+	}
+
+	// deploy tx sender for each portal
+	for _, portal := range portals {
+		err := manager.deployTx(ctx, portal, privateKey)
+		if err != nil {
+			return TxSenderManager{}, errors.Wrap(err, "deploy tx sender")
+		}
+	}
+
+	return manager, nil
+}
+
+func (s TxSenderManager) deployTx(ctx context.Context, portal netman.Portal, privateKey *ecdsa.PrivateKey) error {
```

Comment:
this name isn't correct, this is more `makeInstance` or `newWrapper` or something like that. "Deploy" sounds like "sending transactions"..